src/tests/java/picard/analysis/directed/CollectTargetedMetricsTest.java

```diff
@@ -0,0 +1,160 @@
+package picard.analysis.directed;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMFileWriter;
+import htsjdk.samtools.SAMFileWriterFactory;
+import htsjdk.samtools.SAMReadGroupRecord;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMRecordSetBuilder;
+import htsjdk.samtools.SAMTextHeaderCodec;
+import htsjdk.samtools.metrics.MetricsFile;
+import htsjdk.samtools.util.BufferedLineReader;
+import htsjdk.samtools.util.IOUtil;
+import htsjdk.samtools.util.Log;
+import org.testng.Assert;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+import picard.cmdline.CommandLineProgramTest;
+import picard.sam.SortSam;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Random;
+
+public class CollectTargetedMetricsTest extends CommandLineProgramTest {
+    private final static String sample = "TestSample1";
+    private final static String readGroupId = "TestReadGroup1";
+    private final static String platform = "ILLUMINA";
+    private final static String library = "TestLibrary1";
+
+    private final static File TEST_DIR = new File("testdata/picard/sam/CollectGcBiasMetrics/");
+    private final File dict = new File(TEST_DIR, "Mheader.dict");
+
+    File tempSamFile;
+
+    SAMRecordSetBuilder setBuilder = new SAMRecordSetBuilder(true, SAMFileHeader.SortOrder.coordinate);
+    SAMReadGroupRecord readGroupRecord = new SAMReadGroupRecord(readGroupId);
+
+    @Override
+    public String getCommandLineProgramName() {
+        return CollectTargetedPcrMetrics.class.getSimpleName();
+    }
+
+    //create a samfile with 40000 reads for testing whether a cap is found.
+    @BeforeTest
+    void setupBuilder() throws IOException {
+        final int numReads = 40000;
+        final String readName = "TESTBARCODE";
+
+        tempSamFile = File.createTempFile("CollectTargetedMetrics", ".bam", TEST_DIR);
+        final File tempSamFileUnsorted = File.createTempFile("CollectTargetedMetrics", ".bam", TEST_DIR);
+        tempSamFileUnsorted.deleteOnExit();
+        tempSamFile.deleteOnExit();
+
+        BufferedLineReader bufferedLineReader = null;
+        try {
+            bufferedLineReader = new BufferedLineReader(new FileInputStream(dict));
+        } catch (final FileNotFoundException e) {
+            e.printStackTrace();
+        }
+
+        final SAMTextHeaderCodec codec = new SAMTextHeaderCodec();
+        final SAMFileHeader header = codec.decode(bufferedLineReader, dict.toString());
```

Comment:
Seems like a very complicated way to initialize a samFileHeader...why not the following:

    header.setSequenceDictionary(SAMSequenceDictionaryExtractor.extractDictionary(dict))

?